Week3/homework/3-step3.js

```diff
@@ -2,24 +2,47 @@
 
 // use a 'for' loop
 function repeatStringNumTimesWithFor(str, num) {
-  // add your code here
-  return str;
+
+  let myStr = '';
+
+  for (let i = num; i > 0; i--) {
+    myStr += str;
+  }
+  return myStr;
 }
 
+repeatStringNumTimesWithFor("abc", 3);
+
 console.log('for', repeatStringNumTimesWithFor('abc', 3));
 
 // use a 'while' loop
 function repeatStringNumTimesWithWhile(str, num) {
-  // add your code here
-  return str;
+
+  let myStr = '';
+  while (num > 0) {
+    myStr += str;
+    num--;
+  }
+
+  return myStr;
 }
 
 console.log('while', repeatStringNumTimesWithWhile('abc', 3));
 
+
 // use a 'do...while' loop
 function repeatStringNumTimesWithDoWhile(str, num) {
-  // add your code here
-  return str;
+
+  let myStr = '';
+
+  if (num > 0) {
+    do {
+      myStr += str;
+      num--;
+    } while (num > 0);
+  }
+
+  return myStr;
 }
```

Comment:
👍 Great job.  Something I noticed in some of the other student's work was that they also had a check to make sure num > 0.
I also did not do this, but I thought it was a good idea.